client/src/error.rs

```diff
@@ -29,6 +29,9 @@ pub enum Error {
     /// Invalid transaction
     #[error("Invalid transaction")]
     InvalidTransaction,
+    /// Connected to a node with runtime in incompatible specification version
+    #[error("Connected to a node with runtime in incompatible specification version {0}")]
+    IncompatibleSpecVersion(u32),
```

Comment:
Mentioning the node might confuse people because it sounds like the runtime is something specific to the node. 
```suggestion
    /// Chain is running incompatible runtime spec version
    #[error("Chain is running incompatible runtime spec version {0}")]
    IncompatibleRuntimeVersion(u32),
```

Reply:
Fixed

Reply:
I liked that the variant was explicit about the incompatibility being related to the spec version.